linux/LinuxMachine.c

```diff
@@ -56,13 +56,15 @@ static void LinuxMachine_updateCPUcount(LinuxMachine* this) {
       this->cpuData[1].online = true;
       super->activeCPUs = 1;
       super->existingCPUs = 1;
+      this->maxCoreId = 0;
    }
 
    DIR* dir = opendir("/sys/devices/system/cpu");
    if (!dir)
       return;
 
    unsigned int currExisting = super->existingCPUs;
+   unsigned short maxCoreId = 0;
```

Comment:
Why do we bother with the `unsigned short` data type here (rather than `unsigned int`)? There is no space constraint here (it's not an array) and `short` data types typically process slower in most CPU architectures.

Reply:
To save two bytes in the `CPUData` struct.

Reply:
@cgzones Not worth it when your other code becomes larger due to the instructions processing 16-bit registers. (At least in IA-32 & x86-64, instructions processing 16-bit registers need a prefix and each becomes one byte larger than 32-bit counterparts.)